Add basic application running

phanphare opened the pull request:
Use QProcess to run applications. At the moment, it only allows for one application at a time to be running, and the running application runs in its own thread but can be asynchronously accessed by the launcher. On OS X, it searchs the .app/Contents/Mac OS/ folder for the binary, and on other platforms it simply runs the executable the user has selected.

Commit:
Add basic application running

Adds basic functionality to run an application. The binary that the
user selects (or app bundle on OS X) will be launched. Updates on the
application can be obtained through slots and ProcessChannels (both are
asynchronous). On OS X, it selects the first file in the
.app/Contents/MacOS, which is typically the binary.

lolzballs commented on Source/Library.cpp:
```diff
@@ -23,18 +29,32 @@ void Library::on_testLaunch_clicked()
+        runningProcess->waitForStarted();
+        processRunning = true;
+    } else {
+        QMessageBox messageBox(this);
```
Putting the library class as the parent will cause the QPushButtons on the QMessageBox to be styled, which is not what is desired.

lolzballs replied:
Also, you probably want to display the message box before the QFileDialog pops up to select.

Commit:
Fix UX for running processes

Give an error message before asking for a file selection (if
applicable), and don’t style the filepicker/dialog boxes.

phanphare commented:
I've fixed the issues lolzballs pointed out; thanks for the feedback! The syntax error is also now fixed.

Hal9007 commented on Source/Library.cpp:
```diff
@@ -10,31 +11,53 @@ Library::Library(QWidget *parent) :
 
-        runProcess(file);
+    if (!processRunning) {
+        QFileDialog dialog();
```
I'm getting build errors on both the initialization of the QFileDialog and the QMessageBox below, both in which the compiler thinks you're declaring a function "dialog()" with return type QFileDialog. You should specify the constructor by adding pointers to the parent widget in the initialization of each child.

phanphare replied:
Okay, thanks. I'm not sure how I ended up pushing this commit without compiling first...

Commit:
Syntax Fix

After removing the parent (requested for styling reasons) from dialog
boxes, syntax errors crept in. This is now fixed.

Hal9007 commented:
Thanks for fixing that up, phanphare. Build looks good on Windows and Linux.
